src/drawscene.cpp

```diff
@@ -459,21 +459,55 @@ void draw_pageflip_3d_mode(Camera& camera, bool show_hud,
 #endif
 }
 
-void draw_plain(Camera &camera, bool show_hud, Hud &hud,
-		video::IVideoDriver *driver, bool draw_wield_tool,
-		Client &client, gui::IGUIEnvironment *guienv)
+inline int scaledown(int coef, int size)
 {
+	return (size + coef - 1) / coef;
+}
+
+void draw_plain(Camera &camera, bool show_hud,
+		Hud &hud, video::IVideoDriver *driver,
+		scene::ISceneManager *smgr, const v2u32 &screensize,
+		bool draw_wield_tool, Client &client, gui::IGUIEnvironment *guienv,
+		video::SColor skycolor)
+{
+// Undersampling-specific stuff
+	static video::ITexture *image = NULL;
+	static v2u32 last_pixelated_size = v2u32(0, 0);
+	int undersampling = g_settings->getU16("undersampling");
```

Comment:
you should probably validate that this is 0 2 4 or 8, if it breaks at other values